### PR TITLE
Auto-save draft while collecting data and load on app start

### LINKED ISSUE #708: [Data collection] Auto-save draft while collecting data
Store locally to ensure no data is lost even if app is restarted. Restore previous state when app is reloaded.

Remaining tasks

- [ ] Unit tests
- [ ] Load draw area task from draft

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Towards #708

 * Auto saves the current progress when navigating to previous or next task
 * Creates a new table to store the serialized data to local db
 * Stores the id of the generated row in shared preferences
 * During next app load, we check whether the draft id is stored in shared prefs or not and directly load the data submission screen

Remaining task:
- [ ] Load draw area task from draft 
- [ ] Unit tests

Demo: https://github.com/google/ground-android/assets/8918023/121cdde4-e20e-45bc-9f5e-ca5727e83442

@gino-m  PTAL?

### CODE REVIEW

**Comment by gino-m** [comment]
Looks like there a `detekt` errors,

File: ground/src/main/java/com/google/android/ground/ui/home/HomeScreenViewModel.kt [hunk 1]
```diff
@@ -40,6 +48,33 @@ internal constructor(
   // TODO(#1730): Allow tile source configuration from a non-survey accessible source.
   val showOfflineAreaMenuItem: LiveData<Boolean> = MutableLiveData(true)
 
+  fun maybeNavigateToDraftSubmission() =
+    viewModelScope.launch {
```
[the remark this answers is not in the document]
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/home/HomeScreenViewModel.kt]: Done

File: ground/src/main/java/com/google/android/ground/ui/map/gms/GoogleMapsFragment.kt [hunk 2]
```diff
@@ -238,14 +238,22 @@ class GoogleMapsFragment : SupportMapFragment(), MapFragment {
...
+          CameraPosition(
+            cameraPosition.target.toCoordinates(),
+            cameraPosition.zoom,
+            projection.visibleRegion.latLngBounds.toModelObject(),
+          )
         )
-      )
+      }
+    } catch (exception: IllegalStateException) {
+      // TODO(Shobhit): Prevent the following exception
```
[the remark this answers is not in the document]
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/map/gms/GoogleMapsFragment.kt]: Fixed the underlying issue.

**Review by gino-m — APPROVED** [review]
✔️ !

**Pull request merged**